packages/core/src/sync-realtime/index.ts

```diff
@@ -254,12 +276,17 @@ export const createRealtimeSync = (
       });
     }
 
-    localChain.push(syncBlockToLightBlock(block));
+    unfinalizedBlocks.push(syncBlockToLightBlock(block));
 
-    args.onEvent({
+    // Make sure `transactions` can be garbage collected
+    // @ts-ignore
+    block.transactions = undefined;
+
+    await args.onEvent({
```

Comment:
Why did you add the `await` here?

Reply:
[See above response](https://github.com/ponder-sh/ponder/pull/1115#discussion_r1793637042)